### PR TITLE
use custom 'controller' role name if available

### PR DESCRIPTION
**Pull request by laverya**

### CODE REVIEW

File: pkg/addons/embeddedclusteroperator/embeddedclusteroperator.go [hunk 1]
```diff
@@ -76,9 +76,9 @@ func (e *EmbeddedClusterOperator) GenerateHelmConfig() ([]v1beta1.Chart, []v1bet
 	return []v1beta1.Chart{chartConfig}, nil, nil
 }
 
-// readEmbeddedClusterConfig reads and unmarshal the Config object from the embedded cluster
+// ReadEmbeddedClusterConfig reads and unmarshal the Config object from the embedded cluster
 // that has been embedded into this binary through a release.
-func (e *EmbeddedClusterOperator) readEmbeddedClusterConfig() (*embeddedclusterv1beta1.Config, error) {
+func (e *EmbeddedClusterOperator) ReadEmbeddedClusterConfig() (*embeddedclusterv1beta1.Config, error) {
```
Comment (ricardomaraschini) [inline pkg/addons/embeddedclusteroperator/embeddedclusteroperator.go:81]: Couldn't you call `customization.AdminConsole{}.EmbeddedClusterConfig()` directly ?

Comment (laverya) [reply pkg/addons/embeddedclusteroperator/embeddedclusteroperator.go:81]: yeah but this does the parsing and I'd rather just have that in one place

investigating that is what led to my "we expect unsupported overrides to be two different formats" comment in slack

Comment (laverya) [reply pkg/addons/embeddedclusteroperator/embeddedclusteroperator.go]: I've updated `customization.AdminConsole{}.EmbeddedClusterConfig()` to return the parsed version, and removed this function entirely

File: pkg/config/config.go [hunk 2]
```diff
@@ -320,21 +325,10 @@ type UnsupportedConfigOverrides struct {
 }
 
 // ApplyEmbeddedUnsupportedOverrides applies the custom configuration to the cluster config.
-func ApplyEmbeddedUnsupportedOverrides(config *v1beta1.Cluster, embconfig []byte) error {
-	if embconfig == nil {
+func ApplyEmbeddedUnsupportedOverrides(config *v1beta1.Cluster, embconfig string) error {
+	if embconfig == "" {
 		return nil
 	}
-	var overrides UnsupportedConfigOverrides
```
Comment (ricardomaraschini) [inline pkg/config/config.go:327]: I think we should drop this struct, useless stuff.

**Review by ricardomaraschini — DISMISSED** [review]
If you can remove that struct that would be perfect. Not a blocker.

File: pkg/config/config.go (hunk 2 above)
Comment (laverya) [reply pkg/config/config.go:327]: done

**Review by ricardomaraschini — APPROVED** [review]

**Pull request merged**